### PR TITLE
Add support for BoxWebLink

### PR DESCRIPTION
**Pull request by Zeimyth**
This resolves issue #124.

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @Zeimyth, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

**Comment by Zeimyth** [comment]
CLA signed

**Comment by boxcla** [comment]
Verified that @Zeimyth has just signed the CLA. Thanks, and we look forward to your contribution.

File: src/test/java/com/box/sdk/BoxWebLinkTest.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        BoxWebLink uploadedWebLink = rootFolder.createWebLink(originalFileName, url, description).getResource();
+
+        BoxWebLink.Info newInfo = uploadedWebLink.new Info();
+        newInfo.setName(newFileName);
+        uploadedWebLink.updateInfo(newInfo);
+
+        assertThat(newInfo.getName(), is(equalTo(newFileName)));
+
+        uploadedWebLink.delete();
+    }
```
Comment (tarrencev) [inline src/test/java/com/box/sdk/BoxWebLinkTest.java]: Could we add tests for the each of the constructors

**Comment by tarrencev** [comment]
@Zeimyth thanks for the PR! Just left a comment about some tests. Otherwise, this looks good to me

**Comment by Zeimyth** [comment]
@tarrencev Added the extra constructor tests. Sorry, I thought to do that, but then I completely spaced it.

**Comment by gcurtis** [comment]
Thanks! This looks great.

**Pull request merged**